src/main/java/org/broadinstitute/hellbender/tools/spark/bwa/BwaSparkEngine.java

```diff
@@ -121,6 +122,9 @@ public void close() {
             this.bwaMemIndex = BwaMemIndexCache.getInstance(indexFileName);
             this.readsHeader = readsHeader;
             this.alignsPairs = alignsPairs;
+            if ( alignsPairs && readsHeader.getSortOrder() != SAMFileHeader.SortOrder.queryname ) {
+                throw new UserException("Input must be queryname sorted unless you use single-ended alignment mode.");
```

Comment:
Can you add a test that exercises this case?

Reply:
What if it is "unsorted" but with group order "queryname"?

Reply:
@magicDGS That's a very good point.  That should allowed as well.  I'll update it.